src/qt/listctrl.cpp

```diff
@@ -1133,13 +1385,42 @@ int wxListCtrl::GetColumnWidth(int col) const
 
 bool wxListCtrl::SetColumnWidth(int col, int width)
 {
-    if ( width < 0 )
+    const auto header = m_qtTreeWidget->header();
+
+    if ( header &&
+         col == GetColumnIndexFromOrder(col) &&
+         col == GetColumnCount() - 1 )
     {
-        m_qtTreeWidget->resizeColumnToContents(width);
-        return true;
+        // Always stretch the last section if _with_ is either
+        // wxLIST_AUTOSIZE or wxLIST_AUTOSIZE_USEHEADER
+        header->setStretchLastSection( width < 0 );
     }
 
-    m_qtTreeWidget->setColumnWidth(col, width);
+    if ( width >= 0 )
+        m_qtTreeWidget->setColumnWidth(col, width);
```

Comment:
Having braces around `else` but not `if` branch looks confusing, let's use them here too:
```suggestion
    {
        m_qtTreeWidget->setColumnWidth(col, width);
    }
```